src/measure_theory/integral_eq_improper.lean

```diff
@@ -0,0 +1,512 @@
+Authors: Anatole Dedecker
+-/
+import measure_theory.interval_integral
+import order.filter.at_top_bot
+
+/-!
+# Links between an integral and its "improper" version
+
+In its current state, mathlib only knows how to talk about definite ("proper") integrals,
+in the sense that it treats integrals over `[x, +∞)` the same as it treats integrals over
+`[y, z]`. For example, the integral over `[1, +∞)` is **not** defined to be the limit of
+the integral over `[1, x]` as `x` tends to `+∞`, which is known as an **improper integral**.
+
+Indeed, the "proper" definition is stronger than the "improper" one. The usual counterexample
+is `x ↦ sin(x)/x`, which has an improper integral over `[1, +∞)` but no definite integral.
+
+Although definite integrals have better properties, they are hardly usable when it comes to
+computing integrals on unbounded sets, which is much easier using limits. Thus, in this file,
+we prove various ways of studying the proper integral by studying the improper one.
+
+## Definitions
+
+The main definition of this file is `measure_theory.mono_ae_cover`. It is a rather technical
+definition whose sole purpose is generalizing and factoring proofs. For a sequence `φ` of subsets
+of a measurable space `α` equipped with a measure `μ`, one should think of a hypothesis
+`hφ : mono_ae_cover μ φ` as a sufficient condition for being able to interpret
+`∫ x, f x ∂μ` (if it exists) as the limit of `∫ x in φ n, f x ∂μ` as `n` tends to `+∞`.
+
+When using this definition with a measure restricted to a set `s`, which happens fairly often,
+one should not try too hard to use a `mono_ae_cover` of subsets of `s`, as it often makes proofs
+more complicated than necessary. See for example the proof of
+`integrable_on_Iic_of_tendsto_interval_integral_norm` where we use `Ioi`s as a `mono_ae_cover`
+w.r.t. `μ.restrict (Iic b)`.
+
+## Main statements
+
+- `measure_theory.set_lintegral_tendsto_lintegral` : if `φ` is a `mono_ae_cover` and
+  `f` a measurable `ennreal`-valued function, then `∫⁻ x in φ n, f x ∂μ` tends to `∫⁻ x, f x ∂μ`
+  as `n` tends to `+∞`
+- `measure_theory.integrable_of_set_integral_norm_tendsto` : if `φ` is a `mono_ae_cover`,
+  `f` measurable and integrable on each `φ n`, and `∫ x in φ n, ∥f x∥ ∂μ` tends to some
+  `I : ℝ` as n tends to `+∞`, then `f` is integrable
+- `measure_theory.set_integral_tendsto_integral` : if `φ` is a `mono_ae_cover`,
+  `f` measurable and integrable (globally), then `∫ x in φ n, f x ∂μ` tends to `∫ x, f x ∂μ`
+  as `n` tends to `+∞`
+
+We then specialize these lemmas to various use cases involving intervals, which are frequent
+in analysis.
+-/
+
+open measure_theory filter set
+open_locale ennreal nnreal topological_space
+
+namespace measure_theory
+
+section mono_ae_cover
+
+variables {α ι : Type*} [ordered_add_comm_monoid ι]
+  [measurable_space α] (μ : measure α)
+
+/-- A sequence `φ` of subsets of `α` is a `mono_ae_cover` w.r.t. a measure `μ`
```

Comment:
I have a math question. It seems to me that all theorems below on integrals still hold if you remove the assumption `mono`. Am I overlooking a stupid counterexample? Can you point out in which proof you feel this condition is really important?

Reply:
I think you are right, this assumption seems avoidable everywhere. I don't have a deep understanding of measure theory, so I was mostly adding assumptions which were necessary to make my proof work, but I think I see how to avoid this one. 

Thanks for noticing this, I'll change it, and sorry for wasting your time on this :/

Reply:
This took way too long, but the refactor is finally done ! The `monotone` assumption is gone, the whole file now looks a lot cleaner in my opinion, and I learned a lot in the process, so thanks again :+1: